setup.py

```diff
@@ -1,35 +1,52 @@
+#!/usr/bin/env python
+# -*- coding: utf-8; mode: python -*-
 
-import os
+import re
+import codecs
 from setuptools import setup, find_packages
 
-here = os.path.abspath(os.path.dirname(__file__))
+def read_file(fname):
+    with codecs.open(fname, 'r', 'utf-8') as f:
+        return f.read()
 
-with open(os.path.join(here, 'README.rst')) as f:
-    README = f.read()
+def find_meta(meta):
+    """
+    Extract __*meta*__ from META_FILE.
+    """
+    meta_match = re.search(
+        r"^__{meta}__\s*=\s*['\"]([^'\"]*)['\"]".format(meta=meta),
+        META_FILE
+        , re.M )
+    if meta_match:
+        return meta_match.group(1)
+    raise RuntimeError("Unable to find __{meta}__ string.".format(meta=meta))
 
-with open(os.path.join(here, 'CHANGES.txt')) as f:
-    CHANGES = f.read()
+NAME             = 'hawkauthlib'
+REQUIRES         = ['webob']
+EXTRAS_REQUIRE   = {
+    # https://github.com/pypa/pip/issues/1197
+    'test'       : [ 'requests' ]
+}
+META_FILE        = read_file('hawkauthlib/__init__.py')
+LONG_DESCRIPTION = [ read_file(n) for n in ['README.rst', 'CHANGES.txt']]
 
-requires = ['webob']
 
-tests_requires = requires + ['requests']
-
-setup(name='hawkauthlib',
-      version='0.1.1',
-      description='hawkauth',
-      long_description=README + '\n\n' + CHANGES,
-      license='MPLv2.0',
-      classifiers=[
-        "Programming Language :: Python",
-        "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
-        ],
-      author='Mozilla Services',
-      author_email='services-dev@mozilla.org',
-      url='https://github.com/mozilla-services/hawkauthlib',
-      keywords='authentication Hawk HTTP request signing',
-      packages=find_packages(),
-      include_package_data=True,
-      zip_safe=False,
-      install_requires=requires,
-      tests_require=tests_requires,
-      test_suite="hawkauthlib")
+setup(name                   = NAME
+      , version              = find_meta('version')
```

Comment:
I'm not a fan of this leading-comma style, and I think it'd be a bit odd to use it here but not refactor the rest of the code to also use it.

Reply:
Ha .. I always like to place operators in front, but I'am also aware that I make myself unpopular with :) ... I think you are right; no leading-comma style in this project / I will fix it.